spectrum-network/src/peer/types.rs

```diff
@@ -0,0 +1,26 @@
+/// Opaque identifier for an incoming connection. Allocated by the network.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct IncomingIndex(pub u64);
```

Comment:
Better to hid internal type. If it's needed to return it, then better follow `into_inner(self)` "convention"